tgui/packages/tgui/interfaces/PowernetDebugger.js

```diff
@@ -0,0 +1,371 @@
+import { useBackend, useLocalState } from '../backend';
+import { Button, Section, Table, Box, LabeledList, Collapsible, ProgressBar, Tabs, Icon } from '../components';
+import { Window } from '../layouts';
+
+export const PowernetDebugger = (props, context) => {
+  return (
+    <Window resizable width={1000} height={750}>
+      <Window.Content scrollable>
+        <DebuggerNavigation />
+        <DebuggerContent />
+      </Window.Content>
+    </Window>
+  );
+};
+
+const DebuggerNavigation = (properties, context) => {
+  const { data, act } = useBackend(context);
+  const [tabIndex, setTabIndex] = useLocalState(context, 'tabIndex', 'powernets');
+  const { selected_nets } = data;
+  return (
+    <Tabs>
+      <Tabs.Tab
+        icon="list"
+        selected={'powernets' === tabIndex}
+        onClick={() => {
+          setTabIndex('powernets');
+          act('set_page', { 'page': 1 });
+        }}
+      >
+        Global Powernet List
+      </Tabs.Tab>
+      {selected_nets.map((net) => (
+        <Tabs.Tab
+          key={net}
+          selected={net === tabIndex}
+          onClick={() => {
+            act('detailed_view', { 'PW_UID': net });
+            setTabIndex(net);
+          }}
+        >
+          <Icon name="bolt" />
+          {net}
+        </Tabs.Tab>
+      ))}
+      <Tabs.Tab color="red" onClick={() => act('clear_tabs')}>
+        <Icon name="folder-minus" />
+        Clear Tabs
+      </Tabs.Tab>
+    </Tabs>
+  );
+};
+
+const DebuggerContent = (props, context) => {
+  const { data } = useBackend(context);
+  const { debug_page } = data;
+
+  switch (debug_page) {
+    case 1:
+      return <PowernetList />;
+    case 2:
+      return <DetailedPowernet />;
+    default:
+      return "You are somehow on a tab that doesn't exist! Please let a coder know.";
+  }
+};
+
+const PowernetList = (properties, context) => {
+  const [tabIndex, setTabIndex] = useLocalState(context, 'tabIndex');
+  const { act, data } = useBackend(context);
+
+  const { powernets, filters } = data;
+
+  return (
+    <Section title="Powernets">
+      <LabeledList>
+        <LabeledList.Item label="Filters">
+          <Button.Checkbox
+            content="Off-Station"
+            checked={!filters.off_station}
+            selected={!filters.off_station}
+            onClick={() => act('filter_off_station')}
+          />
+        </LabeledList.Item>
+      </LabeledList>
+      <Table className="Library__Booklist">
```

Comment:
I'm fine with reusing the library computer CSS here but could you add a TODO somewhere to consolidate this class somewhere? it has similar copies in AccountsUplinkTerminal, MedicalRecords, PdaPainter, and SecurityRecords